### ISSUE TITLE
Store PopGameID doesn't pop - rename to peek

### ISSUE THREAD

**Issue by torbensky** [battlesnakeio/roadmap#100]
The documentation for the `Store` interface doesn't mention that a game should be removed from the available set. The inmem reference implementation also doesn't remove a game from the available set. Because of this, the operation is actually what would be called a "peek" operation in normal queue/stack terminology.

For this reason, I am suggesting we should make it clearer by renaming `PopGameID` to `PeekGameID`.

I do not think we should actually "pop" as the name suggests either, because that complicates the store operation and increases the burden on each store implementation. I think the store operations should be kept as very simple, primitive operations that are coordinated by controller logic for more complex "transactions".

So TLDR;
rename `PopGameID` to `PeekGameID` keeping the behavior the same as it is currently

**Comment by ColDog** [comment]
I'm always in favour of renames if it is more clear. Just going through The reason that it is named `PopGameID` is because it pops a game off of a queue that is unlocked games. Implementation details for each store represent this queue in not necessarily a queue like data structure.
